From fa520d9076806060ca103844f587462bd38981f7 Mon Sep 17 00:00:00 2001
From: Rio Martinez <rio.martinez@remarkable.com>
Date: Wed, 18 Jun 2025 17:07:49 +0200
Subject: [PATCH] chore: wasi support

This PR adds rudimentary WebAssembly/WASI support
by excluding unsupported calls to non-implemented
WASI/posix syscalls.

This also adds an OS definition for WASI in Qt.
---
 src/corelib/global/qsystemdetection.h | 2 ++
 src/gui/CMakeLists.txt                | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/corelib/global/qsystemdetection.h b/src/corelib/global/qsystemdetection.h
index 0cbcef27407..0dfb4f0070a 100644
--- a/src/corelib/global/qsystemdetection.h
+++ b/src/corelib/global/qsystemdetection.h
@@ -130,6 +130,8 @@
 #elif defined(__HAIKU__)
 #  define Q_OS_HAIKU
 #elif defined(__MAKEDEPEND__)
+#elif defined(__wasi__)
+#  define Q_OS_WASI
 #else
 #  error "Qt has not been ported to this OS - see http://www.qt-project.org/"
 #endif
diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index 2cc72625131..02fb20d9e96 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -992,7 +992,7 @@ qt_internal_extend_target(Gui CONDITION UNIX
         platform/unix/qunixnativeinterface.cpp
 )
 
-qt_internal_extend_target(Gui CONDITION UNIX AND NOT WASM
+qt_internal_extend_target(Gui CONDITION UNIX AND (NOT WASM OR WASI)
     SOURCES
         platform/unix/qgenericunixeventdispatcher.cpp platform/unix/qgenericunixeventdispatcher_p.h
         platform/unix/qunixeventdispatcher.cpp
-- 
2.39.5 (Apple Git-154)

